app/waf/src/controllers/wafpolicy.controller.ts

```diff
@@ -120,6 +129,94 @@ export class WafpolicyController extends BaseController {
     );
   }
 
+  @post(prefix + '/wafpolicies/{id}/adcs/{adcId}', {
+    responses: {
+      '204': Schema.emptyResponse('Uploading WAF Policy resource'),
+      '404': Schema.notFound('Can not find trustedDeivceId'),
+      '422': Schema.unprocessableEntity(' Adc is not trusted'),
+    },
+  })
+  async wafpolicyUpload(
+    @param(Schema.pathParameter('id', 'WAF Policy resource ID')) id: string,
+    @param(Schema.pathParameter('adcId', 'ADC resource ID')) adcId: string,
+  ): Promise<void> {
+    let wafpolicy: Wafpolicy = await this.wafpolicyRepository.findById(id, {
+      where: {
+        and: [
+          {or: [{tenantId: await this.tenantId}, {public: true}]},
+          {id: id},
+        ],
+      },
+    });
+
+    let adc: Adc = await this.adcRepository.findById(adcId, undefined, {
+      tenantId: await this.tenantId,
+    });
+
+    if (adc.trustedDeviceId === undefined) {
+      throw new HttpErrors.UnprocessableEntity(`Adc: ${adc.id} is not trusted`);
+    }
+
+    // TODO: ADC and wafpolicy are many-to-many relationship,
+    // relationship check need here in some day.
+
+    await this.asgMgr.wafpolicyUploadByUrl(
+      wafpolicy.url,
+      adc.trustedDeviceId,
+      wafpolicy.id,
+    );
+  }
+
+  @get(prefix + '/wafpolicies/{id}/adcs/{adcId}', {
+    responses: {
+      '200': 'Status of wafpolicy',
+      '404': Schema.notFound('Can not find relative resource'),
+      '422': Schema.unprocessableEntity(' Adc is not trusted'),
+    },
+  })
+  async wafpolicyCheckStatus(
+    @param(Schema.pathParameter('id', 'WAF Policy resource ID')) id: string,
+    @param(Schema.pathParameter('adcId', 'ADC resource ID')) adcId: string,
+  ): Promise<Response> {
+    let wafpolicy: Wafpolicy = await this.wafpolicyRepository.findById(id, {
+      where: {
+        and: [
+          {or: [{tenantId: await this.tenantId}, {public: true}]},
+          {id: id},
+        ],
+      },
+    });
+
+    // TODO: there is no many-to-many relationship for now
+
+    let adc: Adc = await this.adcRepository.findById(adcId, undefined, {
+      tenantId: await this.tenantId,
+    });
+
+    if (adc.trustedDeviceId === undefined) {
+      throw new HttpErrors.UnprocessableEntity(`Adc: ${adc.id} is not trusted`);
+    }
+
+    // TODO: ADC and wafpolicy are many-to-many relationship,
+    // relationship check need here in some day.
+
+    let resp = await this.asgMgr.wafpolicyCheckByName(
```

Comment:
There is no `try ... catch ...` surrounding this call to service. Also there's no error handling in function `wafpolicyCheckByName`. It would report 500 error if service returns non-200*.